### DISCUSSION TITLE
[Nuxt 3] Can't seem to make RxJS work

### DISCUSSION THREAD

**Discussion by Jprnp** [nuxt/nuxt#16539] [Questions]
I'm developing an application for study purposes with **Nuxt (3.0.0-rc.11)** and for a feature a wanted to use RxJS.
I installed **RxJS (7.5.6)** through 'npm i rxjs' and it installs fine. I can even import the operators and everything in my <script setup>, but when I instantiate anything, ( **'new Subject()'** ), and load the page it's in, the following error pops up on both backend and frontend consoles:

```
[nuxt] [request error] [unhandled] [500] Cannot find module './internal/Observable'
Require stack:
- /home/jprnp/Nuxt Projects/nuxt-test/rxjs
  at Module._resolveFilename (node:internal/modules/cjs/loader:955:15)  
  at Module._load (node:internal/modules/cjs/loader:803:27)  
  at Module.require (node:internal/modules/cjs/loader:1021:19)  
  at require (node:internal/modules/cjs/helpers:103:18)  
  at rxjs:17:20  
  at ViteNodeRunner.directRequest (/home/jprnp/Nuxt%20Projects/nuxt-test/node_modules/vite-node/dist/client.mjs:251:11)  
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)  
  at async ViteNodeRunner.cachedRequest (/home/jprnp/Nuxt%20Projects/nuxt-test/node_modules/vite-node/dist/client.mjs:128:12)  
  at async request (/home/jprnp/Nuxt%20Projects/nuxt-test/node_modules/vite-node/dist/client.mjs:151:16)  
  at async ./components/Test.vue:1:238
```

Am I forgetting to do any additional config to use RxJS? I tried so many things by now, I'm almost giving up on RxJS in this project. And no one seems to be talking about it anywhere.

**Comment by sevillaarvin** [comment]
Hi, try to install the esm bundle. Well first uninstall the existing rxjs with `npm uninstall rxjs`, then `npm install rxjs@npm:@esm-bundle/rxjs`.

The problem is that `rxjs` uses the **cjs** bundle, even though there is an **esm** bundle as well. I can't figure out how to use esm and not cjs in the normal `rxjs` package, but forcing rxjs to only contain esm by specifically installing it should do the trick.

That said, I don't really see the point of using rxjs in Vue. Vue already has a reactivity system, and adding rxjs on top of that would be redundant in my view. If I may raise my own question, what's the benefit of using rxjs on top of vue, react, etc.?

References:

https://github.com/ReactiveX/rxjs/issues/4416#issuecomment-620847759

https://www.npmjs.com/package/@esm-bundle/rxjs

**Reply by Jprnp** [reply]
Hey, thank you so much for your answer. It works like a charm!

Answering your question: i'm used to work with the RxjS from my background in Angular projects that use those extensively, and I like the convenience of things like Subjects combined with operators like debounceTime, distinctUntilChanged, combineLatest, etc. As I am quite new to Nuxt/Vue, I don't know alternatives to these so I can incorporate in my toolbelt. But maybe I'm really just making things more difficult for me hahahah.

Thanks again!

**Reply by NexZhu** [reply]
This package does not contains TypeScript types, so not for TS users.
Another way is to add `rxjs` to `build.transpile` in `nuxt.config.ts`.

**Reply by gyhyfj** [reply]
> This package does not contains TypeScript types, so not for TS users. Another way is to add `rxjs` to `build.transpile` in `nuxt.config.ts`.

I did you said, but is still not wokr:
```ts
  build: {
    transpile: ['konva'],
  },
```

**Reply by JasonLandbridge** [reply]
@gyhyfj this works:

```typescript
// nuxt.config.ts
export default defineNuxtConfig({
  build: {
    transpile: ["rxjs"],
  },
});
```

**Reply by gyhyfj** [reply]
@JasonLandbridge what's your nuxt version?

**Reply by JasonLandbridge** [reply]
The latest as of now, so `3.6.3`
